client/post-editor/post-editor.jsx

```diff
@@ -574,10 +578,16 @@ export const PostEditor = React.createClass( {
 		this.setState( { isSaving: true } );
 	},
 
-	onPreview: function( event ) {
+	onPreview: function( action, event ) {
 		var status = 'draft',
 			previewPost;
 
+		if ( this.state.previewAction !== action ) {
+			this.setState( {
+				previewAction: action
+			} );
+		}
```

Comment:
Right - we could use a `setTimeout` to eventually revert the state back to `null`, but that still feels a bit hacky. Outside of closing the preview window, there isn't a great cue to reset the `this.state.previewAction` back to null :(